packages/payload/src/admin/components/utilities/Preferences/index.tsx

```diff
@@ -65,14 +67,57 @@ export const PreferencesProvider: React.FC<{ children?: React.ReactNode }> = ({
   )
 
   const setPreference = useCallback(
-    async (key: string, value: unknown): Promise<void> => {
-      preferencesRef.current[key] = value
-      await requests.post(
-        `${serverURL}${api}/payload-preferences/${key}`,
-        requestOptions(value, i18n.language),
-      )
+    async (key: string, value: unknown, merge = false): Promise<void> => {
+      if (merge === false) {
+        preferencesRef.current[key] = value
+        await requests.post(
+          `${serverURL}${api}/payload-preferences/${key}`,
+          requestOptions(value, i18n.language),
+        )
+        return
+      }
+
+      let newValue = value
+      const currentPreference = await getPreference(key)
+      if (
+        typeof value === 'object' &&
+        typeof preferencesRef.current[key] === 'object' &&
+        typeof newValue === 'object'
+      ) {
+        newValue = { ...(currentPreference || {}), ...value }
+        if (isDeepEqual(newValue, currentPreference)) {
+          return
+        }
+        pendingUpdate.current[key] = {
+          ...pendingUpdate.current[key],
+          ...(newValue as Record<string, unknown>),
+        }
+      } else {
+        if (newValue === currentPreference) {
+          return
+        }
+        pendingUpdate.current[key] = newValue
+      }
+
+      const updatePreference = async () => {
+        if (isDeepEqual(pendingUpdate.current[key], preferencesRef.current[key])) {
+          return
+        }
+        preferencesRef.current[key] = pendingUpdate.current[key]
+
+        await requests.post(
+          `${serverURL}${api}/payload-preferences/${key}`,
+          requestOptions(preferencesRef.current[key], i18n.language),
+        )
+        delete pendingUpdate.current[key]
+      }
+
+      // use timeout to allow multiple changes of different values using the same key in one request
+      setTimeout(() => {
```

Comment:
Should we add a time to the timeout? say 50ms? Makes it a bit more readable imo

Reply:
This could break e2e testing to add a ms timeout. It does look odd, but given the comments I think it is okay to leave it out.